src/.nuget/packageLoad.targets

```diff
@@ -0,0 +1,156 @@
+  -->
+
+  <!-- These properties are related to NuGet -->
+  <PropertyGroup>
+    <NuGetToolPath Condition="'$(NuGetToolPath)'==''">$([System.IO.Path]::Combine($(NuGetDir), "NuGet.exe"))</NuGetToolPath>
+    <NuGetConfigFile>$([System.IO.Path]::Combine($(NuGetConfigDir),"NuGet.Config"))</NuGetConfigFile>
+    <NuGetPackagesFile>$([System.IO.Path]::Combine($(NuGetConfigDir),"packages.config"))</NuGetPackagesFile>
+    <NuGetPackagesFileOSSpecific>$([System.IO.Path]::Combine($(NuGetConfigDir),"packages.$(OsEnvironment).config"))</NuGetPackagesFileOSSpecific>
+    <NuGetProjectJsonFile>$([System.IO.Path]::Combine($(NuGetConfigDir),"project.json"))</NuGetProjectJsonFile>
+    <NuGetProjectLockJsonFile>$([System.IO.Path]::Combine($(NuGetConfigDir),"project.lock.json"))</NuGetProjectLockJsonFile>
+    <NuGetCommonOptions>-ConfigFile &quot;$(NuGetConfigFile)&quot; -Verbosity detailed</NuGetCommonOptions>
+    <NuGetPackageInstallOptions>-OutputDirectory &quot;$(NuGetDir.TrimEnd('\').TrimEnd('\'))&quot;</NuGetPackageInstallOptions>
+    <NuGetRestoreCommand>$(NuGetToolPath) restore $(NuGetCommonOptions)</NuGetRestoreCommand>
+    <SolutionSemaphore>$(NuGetDir)Solution.semaphore</SolutionSemaphore>
+    <NugetDownloadURL>https://dist.nuget.org/win-x86-commandline/v3.2.0/nuget.exe</NugetDownloadURL>
+    <DownloadCommand Condition="'$(OsEnvironment)'=='Windows_NT'">powershell -noprofile -nologo -command "(new-object System.Net.WebClient).DownloadFile('$(NugetDownloadURL)', '$(NuGetToolPath)')"</DownloadCommand>
+    <DownloadCommand Condition="'$(OsEnvironment)'!='Windows_NT'">curl -sSL --create-dirs -o $(NuGetToolPath) $(NugetDownloadURL)</DownloadCommand>
+    <BuildToolsSemaphore>$(NuGetDir)BuildTools.semaphore</BuildToolsSemaphore>
+  </PropertyGroup>
+
+  <!-- These are properties for the dnu -->
+  <PropertyGroup>
+    <PackagesCache Condition="'$(OsEnvironment)'=='Windows_NT'">$([System.IO.Path]::Combine($(USERPROFILE), '.nuget', 'packages'))</PackagesCache>
+    <PackagesCache Condition="'$(OsEnvironment)'!='Windows_NT'">$([System.IO.Path]::Combine($(HOME), '.nuget', 'packages'))</PackagesCache>
+    <SetDnxPackages Condition="'$(OsEnvironment)'=='Windows_NT'">set DNX_PACKAGES=$(PackagesCache) &amp;</SetDnxPackages>
+    <SetDnxPackages Condition="'$(OsEnvironment)'!='Windows_NT'">export DNX_PACKAGES=$(PackagesCache) ;</SetDnxPackages>
+    <DnuRestoreCommand>$(SetDnxPackages) $(DnuToolPath) restore</DnuRestoreCommand>
+  </PropertyGroup>
+
+  <ItemGroup>
+     <ProjectJsonFile Include="$(SolutionRoot)**\project.json" Exclude="$(NuGetConfigDir)\project.json" />
+     <ProjectJsonFile Include="$(SolutionRoot)**\project.json" />
+  </ItemGroup>
+
+  <!-- Restore NuGet.exe and use it to process Nuget.config. -->
+  <Target Name="_RestoreBuildToolsNugetConfig"
+          Inputs="$(GlobalPropertiesFile);$(NuGetPackagesFile)"
+          Outputs="$(BuildToolsSemaphore);$(BuildToolsSemaphore)"
```

Comment:
I think this should be

    Outputs="$(BuildToolsSemaphore);$(NuGetToolPath)"

Instead of:

    Outputs="$(BuildToolsSemaphore);$(BuildToolsSemaphore)"

Otherwise the target is skipped because it thinks everything is up to date and NuGet.exe isn't downloaded.